src/meta/src/hummock/manager/mod.rs

```diff
@@ -2355,6 +2356,9 @@ async fn write_exclusive_cluster_id(
     let metadata = object_store.list(&cluster_id_dir).await?;
 
     if metadata.is_empty() {
+        if let risingwave_object_store::object::ObjectStoreImpl::S3(s3) = object_store.as_ref() && !do_not_config_object_storage_lifecycle{
+            s3.inner().configure_bucket_lifecycle().await;
+        }
```

Comment:
I think we should move this logic from inside `write_exclusive_cluster_id` to after it.

https://github.com/risingwavelabs/risingwave/blob/5ad3a2e35ff708ba6f97b2f0026cd6f26303ebf3/src/meta/src/hummock/manager/mod.rs#L304-L312